src/CTA.Rules.Models/Tokens/ExpressionToken.cs

```diff
@@ -0,0 +1,17 @@
+﻿using CTA.Rules.Config;
+
+namespace CTA.Rules.Models.Tokens
+{
+    public class ExpressionToken : NodeToken
+    {
+        public override bool Equals(object obj)
+        {
+            var token = (ExpressionToken)obj;
+            return token.Key == this.Key && token.Namespace == this.Namespace && token.Type == this.Type;
+        }
+        public override int GetHashCode()
+        {
+            return 3 * Key.GetHashCode() + Utils.GenerateHashCode(11, this.Namespace) + Utils.GenerateHashCode(13, Type);
```

Comment:
This can be simplified with Hashcode.Combine (This is not consistent in the solution and we might need to change all of them at a later stage to use Hashcode.Combine)

Reply:
No problem, I will make sure at least the other Tokens objects are consistent and test everything.

Reply:
Never mind, it is consistent everywhere now :-)